scubainit/src/util.rs

```diff
@@ -71,10 +71,24 @@ pub fn pop_env_uint(name: &str) -> Result<Option<u32>> {
     };
     let value: u32 = value_str
         .parse()
-        .context(format!("Parsing integer variable {name}=\"{value_str}\""))?;
+        .with_context(|| format!("Parsing decimal integer variable {name}=\"{value_str}\""))?;
     Ok(Some(value))
 }
 
+/// Gets and unsets an octal uint environment variable `name`, or Ok(None) if it is not set.
+///
+/// # Errors
+///
+/// This function will return an error if the variable is set, but is not a valid integer string.
+pub fn pop_env_octal(name: &str) -> Result<Option<u32>> {
+    let value_str = match pop_env_str(name) {
+        None => return Ok(None),
+        Some(s) => s,
+    };
```

Comment:
> Reading Rust is becoming as much fun for me as reading Perl!

Time to embrace pattern matching ;) You could use [let-else](https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html) here.

```suggestion
    let Some(value_str) = pop_env_str(name) else {
        return Ok(None);
```

The previous `pop_env_uint` could use the same pattern.

Reply:
Ah, I do love `let-else` these days, but I was very new to Rust when I first wrote this file.

We can leave this as-is for consistency for and I may do an update later.